src/controllers/profileController.js

```diff
@@ -0,0 +1,104 @@
+import models from '../models';
+import { errorResponse, successResponse } from '../utils';
+
+const { Profiles, Departments, Users } = models;
+/**
+ * It contains the controller methods that handle requests to the profile
+ * endpoint entity
+ * @class profileController
+*/
+export default class profileController {
+/**
+   * It receives the social provider to use for authentication and returns an express
+   * controller function to handle the callback request that happens after an
+   * authentication attempt has happened
+   * @function callback
+   * @param {Object} req The express request object
+   * @param {Object} res The express response object
+   * @returns {Object} UserProfile The current user profile
+*/
+  static async getProfile(req, res) {
+    const { params: { userId } } = req;
+    try {
+      const { dataValues: data } = await Profiles.findOne({
+        where: { userId },
+        include: [
+          {
+            model: Departments,
+            as: 'theDepartment',
+            attributes: [['name', 'department']]
+          },
+          {
+            model: Users,
+            as: 'theUser',
+            attributes: ['firstName', 'lastName', 'email', 'isVerified']
+          }
+        ]
+      });
+      return successResponse(res, 200, 'Profile fetched successfully', data);
+    } catch (error) {
+      return errorResponse(res, 500, 'Operation not successful. Please try again later.');
+    }
+  }
+  
+  /**
+     * It receives the social provider to use for authentication and returns
+     * an express controller function that handles the initial request that
+     * seeks authentication from the social platforms
+     * @function authenticate
+     * @param {Object} req The express request object
+     * @param {Object} res The express response object
+     * @returns {Object} UserProfile The new profile just created
+  */
+  static async addProfile(req, res) {
+    try {
+      const {
+        body: {
+          department: name, phoneNumber, role, jobDescription, imageUrl
+        }
+      } = req;
+      const { userId } = req;
+      const { dataValues: { id: departmentId } } = await Departments.findOne({ where: { name } });
+      const { dataValues: data } = await Profiles.create({
+        userId, phoneNumber, departmentId, role, jobDescription, imageUrl
+      });
+      return successResponse(res, 201, 'Profile created successfully', data);
+    } catch (error) {
+      return errorResponse(res, 500, 'Operation not successful. Please try again later.');
+    }
+  }
+  
+  /**
+   * It receives the social provider to use for authentication and returns
+   * an express controller function that handles the initial request that
+   * seeks authentication from the social platforms
+   * @function authenticate
```

Comment:
Yeah. The documentation is not accurate currently. It's what I will be working on this morning. These texts were just put there to prevent eslint from `shouting`.